hack/build-images.sh

```diff
@@ -1,14 +1,75 @@
 #!/bin/bash
 
-set -ex
+set -o errexit
+set -o nounset
+set -o pipefail
 
-prefix=${PREFIX:-${1:-openshift/origin-}}
-version=${VERSION:-${2:-latest}}
-for component in fluentd elasticsearch kibana deployment curator ; do
-    docker build -t "${prefix}logging-${component}:${version}"       ../$component/
+STARTTIME=$(date +%s)
+
+prefix="${PREFIX:-openshift/origin-}"
+version="${VERSION:-latest}"
+os_tag="${OS_PUSH_TAG:-latest}"
+
+source_root=$(dirname "${0}")/..
+
+# update the FROM statement in the Dockerfile to be a specific
+# tag.  Assumes there is only 1 FROM but will change them all
+# if there is more than one
+function create_dockerfile_using_base_tag {
+  echo "Creating a temp dockerfile and updating the base image..."
+  source=$1
+  tag=$2
+  file=$3
+  while IFS='' read -r line || [[ -n "$line" ]]; do
+      regex="^\s*(FROM.*)(:)(.*)\s*"
+      if [[ $line =~ $regex ]]; then
+         from=${BASH_REMATCH[1]}:$tag
+         echo "Found FROM statement. updating to $from"
+         echo 
+         echo $from >> $file
+      else
+        echo $line >> $file
+      fi
+      
+  done < $source
```

Comment:
I think you could replace this with:

    sed -e "s/^\([:blank:]*FROM[^:]*:\).*/\1${tag}/" $source > $file